pkg/istio-agent/sds-agent.go

```diff
@@ -295,8 +287,7 @@ func (sa *Agent) newWorkloadSecretCache() (workloadSecretCache *cache.SecretCach
 
 	// TODO: this should all be packaged in a plugin, possibly with optional compilation.
 	log.Infof("sa.serverOptions.CAEndpoint == %v", sa.secOpts.CAEndpoint)
-	if (sa.secOpts.CAProviderName == "GoogleCA" || strings.Contains(sa.secOpts.CAEndpoint, "googleapis.com")) &&
-		stsclient.GKEClusterURL != "" {
```

Comment:
Why do we remove this check? Is this a bug?

Reply:
Removing it in this PR is optional (will be required later for VM support). I just think CAProviderName or CAEndpoint is already enough to say that CA plugin is needed. What do you think?

Reply:
It was added because the code was error-looping if GKEClusterURL was not set - whatever APIs were used were failing if that param was missing. I don't think VMs will work without that either - did you test ?